### DISCUSSION TITLE
Active ballast system

### DISCUSSION THREAD

**Discussion by QusayHawari** [OpenFAST/openfast#1139] [Q&A]
Hi, 

I am trying to compare the closed loop response of the 15MW floater with the responses found in :

https://wes.copernicus.org/articles/6/867/2021/ 

The simulation conditions I am comparing my simulation with are for:

- ActiveFloat semi-sub 15MW model.
- NTM with a mean of 10.5m/s
- ESS with 5.11m and a Tp of 9s 

The responses for the above simulation settings are found in Fig.12 pg. 878.

The simulations I get when running the ROSCO controller are different than in the paper. 
The only simulation setting I am missing is adding an active ballast system as it was added in the paper. I've been skimming through the input files, but I cannot seem to find it. 

Thanks,
Qusay.

**Comment by jjonkman** [comment]
Dear @QusayHawari,

The HydroDyn module of OpenFAST supports fixed water ballast and the structural control (StC) submodel of the ServoDyn module of OpenFAST supports various structural controllers such as TMDs and TLDCs, but OpenFAST does not currently have direct support for active water ballasting, where water is actively pumped throughout the floating substructure.  Perhaps the authors of the article you referenced customized OpenFAST to add this functionality?

Best regards,

**Reply by QusayHawari** [reply]
Hi,

I see, so the platform mass and inertia includes the ballasts in them? It's just that those values cannot actively change during the simulation?

From what I understood in the paper, I think they added this effect by changing the platform's center of gravity according to the wind speed as described in section 4.1 pg, 871. 


Thanks for your help :)
Qusay.

**Reply by jjonkman** [reply]
Hi @QusayHawari,

Yes, the ballast can't actively change within a simulation in OpenFAST.  Certainly it is possible in OpenFAST to change where the ballast is located in separate simulations.

Best regards,
